interfaces/devdoc/srw_lock_ll_requirements.md

```diff
@@ -0,0 +1,88 @@
+# `srw_lock_ll` requirements
+
+## Overview
+
+`srw_lock_ll` is the implementation of a slim reader writer lock.
+
+## Exposed API
+
+```c
+#define SRW_LOCK_LL_TRY_ACQUIRE_RESULT_VALUES \
+    SRW_LOCK_LL_TRY_ACQUIRE_OK, \
+    SRW_LOCK_LL_TRY_ACQUIRE_COULD_NOT_ACQUIRE, \
+    SRW_LOCK_LL_TRY_ACQUIRE_INVALID_ARGS
+
+typedef void* SRW_LOCK_LL;
+
+MU_DEFINE_ENUM(SRW_LOCK_LL_TRY_ACQUIRE_RESULT, SRW_LOCK_LL_TRY_ACQUIRE_RESULT_VALUES)
+
+MOCKABLE_FUNCTION(, void, srw_lock_ll_init);
+MOCKABLE_FUNCTION(, void, srw_lock_ll_deinit, SRW_LOCK_LL*, srw_lock_ll);
+
+/*writer APIs*/
+MOCKABLE_FUNCTION(, void, srw_lock_ll_acquire_exclusive, SRW_LOCK_LL*, srw_lock_ll);
+MOCKABLE_FUNCTION(, SRW_LOCK_LL_TRY_ACQUIRE_RESULT, srw_lock_ll_try_acquire_exclusive, SRW_LOCK_LL*, srw_lock_ll);
+MOCKABLE_FUNCTION(, void, srw_lock_ll_release_exclusive, SRW_LOCK_LL*, srw_lock_ll);
+
+/*reader APIs*/
+MOCKABLE_FUNCTION(, void, srw_lock_ll_acquire_shared, SRW_LOCK_LL*, srw_lock_ll);
+MOCKABLE_FUNCTION(, SRW_LOCK_LL_TRY_ACQUIRE_RESULT, srw_lock_ll_try_acquire_shared, SRW_LOCK_LL*, srw_lock_ll);
+MOCKABLE_FUNCTION(, void, srw_lock_ll_release_shared, SRW_LOCK_LL*, srw_lock_ll);
+```
+
+### srw_lock_ll_init
+```c
+MOCKABLE_FUNCTION(, void, srw_lock_ll_init, SRW_LOCK_LL*, srw_lock_ll);
+```
+
+`srw_lock_ll_init` initializes a slim reader writer lock.
+
+### srw_lock_ll_deinit
+```c
+MOCKABLE_FUNCTION(, void, srw_lock_ll_deinit, SRW_LOCK_LL*, srw_lock_ll);
+```
+
+`srw_lock_ll_deinit` frees deinitializes the slim reader writer lock.
+
+### srw_lock_ll_acquire_exclusive
+```c
+MOCKABLE_FUNCTION(, void, srw_lock_ll_acquire_exclusive, SRW_LOCK_LL*, srw_lock_ll);
+```
+
+`srw_lock_ll_acquire_exclusive` acquires the slim reader writer lock in exclusive (writer) mode.
```

Comment:
> exclusive (writer) mode [](http://example.com/codeflow?start=71&length=24)

would be nice to have some REferences to SWR_LOCK (the real one in windows documentation) that would explain what "exclusive" / "shared" mode mean.

Reply:
Added reference. I thought of adding one to wikipedia too :-), but stuck with the microsoft one only.